### PR TITLE
Update the CMSIS-NN glue code to support int16 for softmax.

### LINKED ISSUE #43543: Add int16 support to the CMSIS-NN version of softmax
@tensorflow/micro

The softmax reference kernel have been updated with int16 support, and the CMSIS-NN version should to updated with int16 support as well.

### PR DESCRIPTION
**Pull request by jenselofsson**
Add int16 support to the CMSIS-NN version of softmax.
Fixes #43543

### CODE REVIEW

**Comment by jenselofsson** [comment]
Disregard this PR
